packages/core/src/utils/styleUtils/ssrStyles.ts

```diff
@@ -67,23 +85,33 @@ export const detachExperienceStyles = (experience: Experience): string | undefin
     componentVariablesOverwrites?: Record<string, ComponentPropertyValue>;
     patternWrapper?: ComponentTreeNode;
     wrappingPatternIds: Set<string>;
-    patternNodeIdsChain?: string;
+    parentChainArr?: string[];
   }) => {
     // traversing the tree
-    const queue: ComponentTreeNode[] = [];
-
-    queue.push(...componentTree.children);
+    const queue: QueueItem[] = [];
```

Comment:
Changed the queue structure because we want to keep the parent chain. This is added to resolve correct nodeIds chain. Currently Pattern and Nested Pattern SSR styles are not applying correctly to the node while rendering in `CompositionBlock`.

e.g Create a container in an Experience and add a pattern inside that container, You may notice that SSR styles are not applied to pattern nodes.
In the image below, inside <`assembly`> div, the styles are `cfstyles` and not `cf-hash` which means that styles for pattern when wrapped in a container(s) are not applied on SSR.

<img width="1021" alt="image" src="https://github.com/user-attachments/assets/1d7178d4-30b9-439a-856c-9ad355a20d2e" />